pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -142,6 +145,7 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 		if !deployutil.IsTerminatedDeployment(latestDeployment) {
 			return c.updateStatus(config, existingDeployments)
 		}
+
 		return c.reconcileDeployments(existingDeployments, config)
 	}
 	// If the config is paused we shouldn't create new deployments for it.
```

Comment:
The new deployment doesn't have to do anything with cleanup policy since it applies only to older deployments with zero replicas. The idea here is that cleanup policy should work for paused deployments. After looking more closely, someone can end up in this block iff they pause the deployment and bump latestVersion at the same time, something very unlikely to happen. But we still want cleanup policy to work in that case.